packages/create-sitecore-jss/src/templates/angular-xmcloud/src/app/components/navigation/navigation-item.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { LinkField } from '@sitecore-jss/sitecore-jss-angular';
+import { Field } from '@sitecore-jss/sitecore-jss-angular';
+
+export interface NavItemFields {
+  Id: string;
+  DisplayName: string;
+  Title: Field<string>;
+  NavigationTitle: Field<string>;
+  Href: string;
+  Querystring: string;
+  Children: Array<NavItemFields>;
+  Styles: string[];
+}
+
+@Component({
+  selector: 'app-navigation-item',
+  templateUrl: './navigation-item.component.html',
+})
+export class NavigationItemComponent implements OnInit {
+  @Input() navItemFields: NavItemFields;
+  @Input() relativeLevel: number;
+  @Output() childLinkClickEvent: EventEmitter<Event> = new EventEmitter<Event>();
+  cssClasses = '';
+  isActive = false;
+  linkField = {};
+  childrenRelativeLevel = 0;
+  hasChildren = false;
+
+  constructor() {}
+
+  ngOnInit() {
+    this.cssClasses = `${this.navItemFields.Styles.concat('rel-level' + this.relativeLevel).join(
+      ' '
+    )}`;
+    this.linkField = this.getLinkField(this.navItemFields);
+    this.hasChildren = this.navItemFields.Children && this.navItemFields.Children.length != 0;
+    this.childrenRelativeLevel = this.relativeLevel + 1;
+  }
+
+  onClick(event: Event) {
+    this.childLinkClickEvent.emit(event);
+  }
+
+  private getLinkField = (navItemFields: NavItemFields): LinkField => ({
+    value: {
+      href: navItemFields.Href,
+      title: this.getLinkTitle(navItemFields),
+      querystring: navItemFields.Querystring,
+    },
+  });
```

Comment:
I think we should add a fallback to account for undefined scenarios.

Reply:
in the Promo component PR there is an update for handling empty links (it won't render empty <a></a>). We have another work item for adding empty field placeholders in edit mode